Bearded.Graphics/Windowing/Window.cs

```diff
@@ -44,64 +42,60 @@ public void AttachContextToCallingThread()
             }
         }
 
-        private TaskCompletionSource<Void>? initialization;
-
-        private NativeWindowWrapper? window;
+        private NativeWindowWrapper window = null!;
 
         // TODO(#26): rewrite windowing natively and merge with input, instead of relying on OpenTK.Windowing.Desktop
         [Obsolete("Legacy implementation. There is no replacement yet.")]
-        protected NativeWindow NativeWindow => window ??
-            throw new InvalidOperationException($"Window must be initialised before accessing {nameof(NativeWindow)}.");
+        protected NativeWindow NativeWindow => window;
 
         private bool vsync = true;
 
         protected abstract NativeWindowSettings GetSettings();
 
-        public async Task Initialize()
-        {
-            initialization = new TaskCompletionSource<Void>();
 
-            var uiThread = new Thread(runUI);
-            uiThread.Start();
+        public void Run()
+        {
+            createWindowWithDetachedGLContext();
+            startUpdateThread();
+            runMainThreadLoop();
+        }
 
-            await initialization.Task;
+        private void createWindowWithDetachedGLContext()
+        {
+            setupNativeWindow();
+            OnLoad();
+            TriggerResize();
+            window.DetachContextFromCallingThread();
         }
 
-        private void runUI()
+        private void setupNativeWindow()
         {
             window = new NativeWindowWrapper(GetSettings())
             {
-                IsVisible = true
+                IsVisible = true,
             };
             window.Resize += OnResize;
             window.Closing += OnClosing;
+        }
 
-            OnLoad();
-
-            TriggerResize();
-
-            window.DetachContextFromCallingThread();
-
-            initialization!.SetResult(default);
+        private void startUpdateThread()
+        {
+            var updateThread = new Thread(runGL)
+            {
+                Name = "GL Thread",
+            };
+            updateThread.Start();
+        }
 
+        private void runMainThreadLoop()
+        {
             while (window.Exists && !window.IsExiting)
             {
-                window.ProcessEvents();
+                window.ProcessEvents(TimeSpan.FromSeconds(1d / 200).TotalSeconds);
```

Comment:
It sounds like we made the choice here to make the window update at 200 FPS, yet that is not referenced in the PR as a behaviour change. Is this just making an existing default explicit, are there certain reasons to do this, should this be customisable?

Reply:
Before, because of the Sleep() it would poll events (not update) AT MOST 1000 times/second.

Now it will poll AT LEAST ~200 times/second, or more often, since it's event driven, and the timeout is how long it will wait for events if none occur before it returns (just for us to call it again immediately).

We could potentially wait for much longer timeout, or even block entirely until there are events, if it wasn't for the loop exit condition I suppose.

Once we do a bigger rewrite, I'll dig a bit deeper and see what's actually the best way of handling this, but I'm convinced that what we have now is better than before.